crates/oxc_minifier/src/plugins/replace_global_defines.rs

```diff
@@ -157,39 +180,73 @@ impl<'a> ReplaceGlobalDefines<'a> {
+            Some(StaticMemberExpressionOrMeta::StaticMemberExpression(member));
         let mut cur_part_name = &member.property.name;
-
-        for (i, part) in dot_define.parts.iter().enumerate().rev() {
-            if cur_part_name.as_str() != part {
-                return false;
+        let mut has_matched_part = false;
+        let mut i = dot_define.parts.len() - 1;
+        loop {
+            let part = &dot_define.parts[i];
+            let matched = cur_part_name.as_str() == part;
+            if !matched {
+                // Considering import.meta.env.*
+                // ```js
+                // import.meta.env.test // should matched
+                // import.res.meta.env // should not matched
+                // ```
+                // So we use has_matched_part to track if any part has matched.
+
+                if !dot_define.postfix_wildcard || has_matched_part {
+                    return false;
+                }
             }
 
             if i == 0 {
                 break;
             }
 
             current_part_member_expression = if let Some(member) = current_part_member_expression {
-                match &member.object {
-                    Expression::StaticMemberExpression(member) => {
-                        cur_part_name = &member.property.name;
-                        Some(member)
+                match member {
+                    StaticMemberExpressionOrMeta::StaticMemberExpression(member) => {
+                        match &member.object {
+                            Expression::StaticMemberExpression(member) => {
+                                cur_part_name = &member.property.name;
+                                Some(StaticMemberExpressionOrMeta::StaticMemberExpression(member))
+                            }
+                            Expression::MetaProperty(member) => {
+                                cur_part_name = &member.property.name;
+                                Some(StaticMemberExpressionOrMeta::MetaProperty(member))
+                            }
+                            Expression::Identifier(ident) => {
+                                cur_part_name = &ident.name;
+                                None
+                            }
+                            _ => None,
+                        }
                     }
-                    Expression::Identifier(ident) => {
-                        cur_part_name = &ident.name;
+                    StaticMemberExpressionOrMeta::MetaProperty(meta) => {
+                        cur_part_name = &meta.meta.name;
                         None
                     }
-                    _ => None,
                 }
             } else {
                 return false;
             };
+
+            if matched {
+                i -= 1;
+                has_matched_part = true;
+            }
         }
 
-        true
+        has_matched_part
     }
 }
+
+enum StaticMemberExpressionOrMeta<'a> {
```

Comment:
I think it would be easier to split out dot define and `MetaProperty` in the config instead of mixing them here.

Reply:
Not sure if I understand correctly, do you mean add another field `MetaProperty` in https://github.com/oxc-project/oxc/blob/c65e8fc1ad1c5b71b6a16de44dd2246b293b0660/crates/oxc_minifier/src/plugins/replace_global_defines.rs#L40-L43?

Reply:
yes, because it will be used to test against `MetaProperty` only.

Reply:
Do you mean we only support `import.meta.x.*` syntax rather than general memberExpr wildcard postfix match?
like `process.env.*`?

Reply:
Ok

Reply:
We should only support `process.env.*` if there is a requirement, which is none so far.